### DISCUSSION TITLE
Branch protection 규칙 개선

### DISCUSSION THREAD

**Discussion by crowjdh** [QuesDevTeam/Git-Convention#25] [Ideas]
# 개요
최근 오픈소스 PR을 보다가 새로운 branch protection 규칙들을 발견해서 같이 논의하고 추가하면 좋을 것 같아요

![asdf](https://github.com/QuesDevTeam/Git-Convention/assets/4036461/bd6780bb-1238-4ae1-894c-98073871df3a)

위 이미지에 체크된 2가지 규칙인데:
- `리뷰 가능한 최신 커밋`에 대한 리뷰를 요구
- **모든 conversation의 resolve 요구**

# 논의
이 논의를 제안한 이유는 주로 위의 두 번째 내용 때문인데, 전 PR 리뷰할 때 `내가 연 conversation은 대화/리뷰 후 내가 resolve`하는 기준으로 하고 있어요. 반대로 남이 연 conversation은 코멘트 다는 이외의 액션은 하지 않고요. PR리뷰에 대해 각자의 규칙이 있었을텐데 이 참에 각자 의견을 종합하고 하나의 결론으로 이 레포에 추가하면 좋을 것 같아요.

**Comment by crowjdh** [comment]
@DonghyunNa @yeareater @biio-front @dongjuny 혹시나 해서 소환합니다 :)

**Reply by crowjdh** [reply]
@DonghyunNa @yeareater @dongjuny @biio-front 다들 확인 부탁드려요 :)

**Reply by DonghyunNa** [reply]
확인했습니다. 좋은것 같습니다 :)

**Reply by crowjdh** [reply]
혹시 프런트 팀에서 따로 쓰는 규칙이 더 있는지도 궁금해요 @biio-front

**Reply by dongjuny** [reply]
뒤늦게 봤네요.. 최근에 `모든 conversation의 resolve 요구` 설정을 해제해달라고 요청했던 이유는 기존에 중요하거나 기록을 남겨야 할 코멘트는 `resolve` 상태로 놔두기로 했었어서 요청드렸었고, 지금 생각해 보니 중요한 내용이면 `Issues` 또는 `Discussions` 탭에서 작성하는 게 맞는 거 같네요.👍

**Reply by biio-front** [reply]
프론트에서 따로 사용하는 `resolve` 기준은 따로 없어요!
이건 제 개인적인 규칙인거 같은데 전 누가연 대화이든 다른 상대가 더 확인할 필요가 없다고 생각할때 닫습니다. 예를 들어 리뷰해주셨는데 추가로 논의할거없이 받아들이면 그 댓글 resolve하구요, 추가로 논의가 이어져 상대가 확인해야겠다 싶을땐 그냥 열어둡니다.
그런데 동현님이 말한것처럼 대화 시작한 사람이 대화끝났을때 닫는것도 좋은것같아요!

**Comment by crowjdh** [comment]
다들 의견 감사합니다 :) 전반적으로 필요성에 대해서는 공감이나 해결책이 있는 것 같아 보여요. 크게 정리해 보면:

1. 기본적으로는 `오픈한 사람이 resolve`
2. 추가적으로 기록이 필요한 경우는 `unresolved상태 유지` 혹은 `issue/discussion으로 이동`

이런 식인 것 같아요. 전 2번에 대해서는 닫힌 PR 하나하나 찾아보는 것 보다 `issue/discussion으로 이동`하는 방향이 이슈 트래킹 측면에서 좋을 것 같아요.

이 내용 정리해서 레포에 추가 해두면 명문화 돼서 좋을 것 같아요. 다들 의견 알려주세요!

**Reply by DonghyunNa** [reply]
+ branch protection rule 적용되어 있는 구글 스프레드시트도 같이 명문화 해놓으면 좋을 것 같습니다.
